Userland/Applications/Piano/TrackManager.cpp

```diff
@@ -42,8 +42,10 @@ void TrackManager::fill_buffer(FixedArray<DSP::Sample>& buffer)
     // No need to zero the temp buffer as the track overwrites it anyways.
     buffer.fill_with({});
 
+    Threading::MutexLocker locker(m_samples_mutex);
```

Comment:
This is exactly the thing we should not do in audio code. Taking a mutex lock is almost guaranteed to incur a syscall, which will lead to an unpredictable time overhead.